run-checks

```diff
@@ -0,0 +1,181 @@
+#!/bin/bash
+
+# shellcheck disable=SC2145
+
+REPOROOT=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+
+#----------------------------------------------------#
+# PRINTING ROUTINES                                  #
+#----------------------------------------------------#
+
+P_RESET="\e[0m"
+P_RED="\e[31m"
+P_GREEN="\e[32m"
+
+print_newline() {
+	echo -e "$P_RESET"
+}
+
+print_normal() {
+	echo -e "$P_RESET$@"
+}
+
+print_red() {
+	echo -e "$P_RED$@$P_RESET"
+}
+
+print_green() {
+	echo -e "$P_GREEN$@$P_RESET"
+}
+
+print_special() {
+	echo -ne "$P_RESET$@"
+}
+
+#----------------------------------------------------#
+# TEST CONTROL                                       #
+#----------------------------------------------------#
+
+# If started with:
+# VERBOSE=1 ./run-checks
+# Verbose print everything, not only failures
+VERBOSE=${VERBOSE:-0}
+
+# Number of tests that failed
+FAILED=0
+
+# Run a test and fail based on a non-zero exit code
+#
+# Verbose print the failure case unless VERBOSE=1, in
+# which case verbose print everything (useful for CI
+# logs).
+#
+# Argument $1: Test string description
+# Argument $2..$n: Test command line
+#
+test_check_exit_code() {
+	print_special "=> $1"
+	OUTPUT=$(mktemp)
+	v=$(exec 2>&1 && set -x && set -- "${@:2}")
+
+	set +e
+	"${@:2}" > "$OUTPUT" 2>&1
+
+	# shellcheck disable=SC2181
+	if [ $? -ne 0 ]; then
+                print_red "Fail"
+                print_newline
+		print_red "Command: ${v#*--}"
+                print_newline
+		cat "$OUTPUT"
+		print_newline
+                FAILED=$((FAILED + 1))
+	else
+		print_green "Pass"
+		if [ "$VERBOSE" = "1" ]; then
+	                print_newline
+			print_normal "Command: ${v#*--}"
+			print_newline
+			cat "$OUTPUT"
+			print_newline
+		fi
+	fi
+	set -e
+	rm -f "$OUTPUT"
+}
+
+
+# Run a test and fail based on non-zero output lines
+#
+# Verbose print the failure case unless VERBOSE=1, in
+# which case verbose print everything (useful for CI
+# logs).
+#
+# Argument $1: Test string description
+# Argument $2..$n: Test command line
+#
+test_check_no_output() {
```

Comment:
The duplication with `test_check_exit_code` irks me a bit, I'm probably overthinking it but I wonder if this could be a subset of `test_check_exit_code` somehow, I played and one way might be to just not use this helper and instead do something like: 
```
test_check_exit_code "Formatting ... " sh -c 'D=$(gofmt -d .) && test -z "$(echo "$D")" || (echo "${D}" && exit 1)'
```
but that of course if very brittle so probably not a good idea but still wanted to share my thinking.

Reply:
I was hoping to make the test part (last part) of the script easy for maintainers to add tests to. I think i've found a compromise the still hides the complex part in the helper function, without duplication, and broke out the printing part into subfunctions for easier reading. Let me know if that feels better.